_drafts/partially-ordering-time.md

```diff
@@ -0,0 +1,253 @@
+# Time Is Weird
+## or: why do distributed systems and weak memory models look so similar, anyway
+
+(_Epistemic status: I'm just learning about both of these fields, and these are simply my
+observations from studying them and comparing them. If you work in either field and
+something looks wrong, it probably is; let me know!_)
+
+    There's only one hard problem in computer science: recognising that cache invalidation errors
+    are misnamed. They're just off-by-one errors in the time domain.
+        —Unknown
+
+Time is weird.
+
+Time is weird, not just because we insist on trying to [measure it](#url) and [standardise
+it](#url), though that field is of course amazingly horrifying. Time is weird, because
+when we try to think about what it actually _implies_, that a second has passed since one
+second ago.
+
+The best we've been able to come up with about the arrow of time is that it in some way
+tracks things that “happened” _before_ other things that “happened” _after_ them, in the
+sense, that, uh, the later events have access to information? and entropy? from the
+earlier events? Which means, um, something something flow of causality?
+
+Of course that breaks down, at distances comparable to light speed. When you can no longer
+assume that events are instantaneously knowable, time behaves _seriously_ anomalously. Can
```

Comment:
instantaneously knowable from the perspective of... something that perceives at light speed? what about hearing? or touch? or hearing a set of instructions?

I think a good development of just how weird these concepts are should accommodate things other than just sight, since I think that's what makes re-ordering of instructions logical, yet weird. But I'm just thinking aloud.